pkg/network/ebpf/c/prebuilt/tracer.c

```diff
@@ -159,30 +155,30 @@ static __always_inline int read_conn_tuple(conn_tuple_t* t, struct sock* skp, u6
     // Retrieve addresses
     if (check_family(skp, AF_INET)) {
         t->metadata |= CONN_V4;
-        bpf_probe_read(&t->saddr_l, sizeof(__u32), ((char*)skp) + offset_saddr());
-        bpf_probe_read(&t->daddr_l, sizeof(__u32), ((char*)skp) + offset_daddr());
+        if (t->saddr_l == 0) bpf_probe_read(&t->saddr_l, sizeof(u32), ((char*)skp) + offset_saddr());
```

Comment:
We can get rid of the `if(t->xyz == 0)`  checks

Reply:
Those are necessary since in at least one call `t->daddr_l` will already be filled in.